File: backend/app/schemas/health.py
from datetime import datetime
from decimal import Decimal
from pydantic import BaseModel
from typing import Dict, List, Optional


class StrategyHealthOut(BaseModel):
    health_score: Decimal
    drift_score: Decimal
    state: str
    failure_mode: str
    anomaly: str


class StrategyBaselineOut(BaseModel):
    id: int
    strategy_name: str
    expected_sharpe: Decimal
    expected_win_rate: Decimal
    expected_profit_factor: Decimal
    expected_drawdown: Decimal
    expected_trade_frequency: Decimal
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class StrategyHealthLogOut(BaseModel):
    id: int
    strategy_name: str
    rolling_sharpe: Decimal
    rolling_win_rate: Decimal
    rolling_profit_factor: Decimal
    rolling_drawdown: Decimal
    expectancy: Decimal
    health_score: Decimal
    created_at: datetime

    class Config:
        from_attributes = True


class StrategyAlertOut(BaseModel):
    id: int
    strategy_name: str
    alert_level: str
    message: str
    action_taken: str
    created_at: datetime

    class Config:
        from_attributes = True


class StrategyStateHistoryOut(BaseModel):
    id: int
    strategy_name: str
    old_state: str
    new_state: str
    trigger_reason: str
    created_at: datetime

    class Config:
        from_attributes = True

File: frontend/types/health.ts
export type StrategyHealthStatus = {
  health_score: string | number;
  drift_score: string | number;
  state: "ACTIVE" | "WARNING" | "CRITICAL" | "PAUSED" | string;
  failure_mode: string;
  anomaly: "NORMAL" | "ANOMALOUS" | string;
};

export type StrategyBaseline = {
  id: number;
  strategy_name: string;
  expected_sharpe: string | number;
  expected_win_rate: string | number;
  expected_profit_factor: string | number;
  expected_drawdown: string | number;
  expected_trade_frequency: string | number;
  created_at: string;
  updated_at: string;
};

export type StrategyHealthLog = {
  id: number;
  strategy_name: string;
  rolling_sharpe: string | number;
  rolling_win_rate: string | number;
  rolling_profit_factor: string | number;
  rolling_drawdown: string | number;
  expectancy: string | number;
  health_score: string | number;
  created_at: string;
};

export type StrategyAlert = {
  id: number;
  strategy_name: string;
  alert_level: "GREEN" | "YELLOW" | "ORANGE" | "RED" | string;
  message: string;
  action_taken: string;
  created_at: string;
};

export type StrategyStateTransition = {
  id: number;
  strategy_name: string;
  old_state: string;
  new_state: string;
  trigger_reason: string;
  created_at: string;
};
